src/normalize/user.rs

```diff
@@ -161,7 +162,10 @@ impl Handler<Reason> for NormalizationHandler<'_, '_> {
     fn handle_entry_addr(&mut self, addr: Addr, entry: &MapsEntry) -> Result<()> {
         match &entry.path_name {
             Some(PathName::Path(entry_path)) => {
-                let path = if self.map_files {
+                // In case, we want to extract the build_id using the symbolic path,
+                // we have to check that the file still exists on disk
+                let file_stat = util::stat(&entry_path.symbolic_path);
+                let path = if self.map_files || file_stat.is_err(){
                     &entry_path.maps_file
```

Comment:
This `||` conditional seems weird to me, to be honest. And it would mean we'd still have the risk of permission denied error on rare occassions, which isn't great and seems confusing from a user's perspective. I'd rather we use the symbolic path unconditionally. What do you think?

Reply:
That makes sense. There is a test that fails and my first thought was to make it `green` first and then have your thought.
I'll have to change the test when the lib is deleted.

Reply:
done in https://github.com/libbpf/blazesym/pull/823/commits/106cb73c9070195b68b72307c4bfdc5a1dcffcec

>